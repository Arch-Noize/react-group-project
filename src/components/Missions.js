// Missions.js

import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Badge, Button, Table } from 'react-bootstrap';
import { getMissions, joinMission, leaveMission } from '../redux/mission/missionSlice'; // You need to import these action functions.
import '../styles/Missions.module.css';

const Missions = () => {
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);

  return (
    <div className="table">
      <Table striped bordered hover>
        <thead>
          <tr>
            <th style={{ width: '12%' }}>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th style={{ width: '12%' }}> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission, index) => (
            <tr key={mission.id} className={index % 2 === 0 ? 'alternate-row' : ''}>
              <td style={{ fontWeight: '700' }}>{mission.name}</td>
              <td>{mission.description}</td>
              <td className="text-center align-middle">
                {mission.reserved ? (
                  <>
                    <Badge bg="info">Active Member</Badge>
                  </>
                ) : (
                  <>
                    <Badge bg="secondary">NOT A MEMBER</Badge>
                  </>
                )}
              </td>
              <td className="text-center align-middle">
                {mission.reserved ? (
                  <Button variant="outline-danger" onClick={() => dispatch(leaveMission(mission.id))}>
                    Leave Mission
                  </Button>
                ) : (
                  <Button variant="outline-dark" onClick={() => dispatch(joinMission(mission.id))}>
                    Join Mission
                  </Button>
                )}
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default Missions;
