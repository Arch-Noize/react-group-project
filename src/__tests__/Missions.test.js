import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Missions from '../components/Missions';

const mockStore = configureMockStore([]);

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('Loading missions component', () => {
  it('renders correctly', () => {
    const initialState = {
      missions: {
        missions: [
          {
            id: 1,
            name: 'Mission',
            description: 'This is a lengthy description indicating the mission paramaters',
            reserved: false,
          },
        ],
      },
    };

    const store = mockStore(initialState);

    const { tree } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
