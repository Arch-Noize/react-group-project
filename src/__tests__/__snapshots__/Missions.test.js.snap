// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading missions component renders correctly 1`] = `undefined`;
